### PR TITLE
pythonPackages.aws-adfs: init at 0.12.0

### PR DESCRIPTION
**Pull request by bhipple**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg build python2Packages.aws-adfs python3Packages.aws-adfs

File: pkgs/development/python-modules/aws-adfs/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  # Test suite writes files to $HOME/.aws/, or /homeless-shelter if unset
+  HOME = ".";
+
+  checkInputs = [ pytest pytestrunner pytestcov mock ];
+  propagatedBuildInputs = [ lxml boto3 requests click configparser ];
+
+  # Python3 tests fail with UTF-8 issues due to click
+  doCheck = !isPy3k;
```
Comment (bhipple) [inline pkgs/development/python-modules/aws-adfs/default.nix]: Click also has quite a few python3 UTF-8 errors in its own test suite, but it happens to be disabled at the moment.  It looks like they're preparing a click 7.0 which might fix a handful of locale issues, but 6.7 is still the latest one available as of now.

Comment (dotlambda) [reply pkgs/development/python-modules/aws-adfs/default.nix]: You may have luck with adding `glibcLocales` to `checkInputs` and setting `LC_ALL = "en_US.UTF-8"`.

File: pkgs/development/python-modules/aws-adfs/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, fetchPypi, pythonPackages, isPy3k
```
Comment (dotlambda) [inline pkgs/development/python-modules/aws-adfs/default.nix]: no `pythonPackages`

File: pkgs/development/python-modules/aws-adfs/default.nix (hunk 1 above)
Comment (bhipple) [reply pkgs/development/python-modules/aws-adfs/default.nix]: Aha, I was setting `LC_ALL` but passing `locale` to `checkInputs`; thanks for the tip!  It works now.

**Comment by bhipple** [comment]
@GrahamcOfBorg build python2Packages.aws-adfs python3Packages.aws-adfs

**Pull request merged**